Python/pywarpx/_libwarpx.py

```diff
@@ -30,112 +30,294 @@
     MPI = None
     _MPI_Comm_type = ctypes.c_void_p
 
-# --- Is there a better way of handling constants?
-clight = 2.99792458e+8 # m/s
-
-def _get_package_root():
-    '''
-    Get the path to the installation location (where libwarpx.so would be installed).
-    '''
-    cur = os.path.abspath(__file__)
-    while True:
-        name = os.path.basename(cur)
-        if name == 'pywarpx':
-            return cur
-        elif not name:
-            return ''
-        cur = os.path.dirname(cur)
-
-# --- Use geometry to determine whether to import the 1D, 2D, 3D or RZ version.
-# --- This assumes that the input is setup before this module is imported,
-# --- which should normally be the case.
-# --- Default to 3D if geometry is not setup yet.
-try:
-    _prob_lo = geometry.prob_lo
-    _coord_sys = geometry.coord_sys
-except AttributeError:
-    geometry_dim = '3d'
-else:
-    if _coord_sys == 0:
-        geometry_dim = '%dd'%len(_prob_lo)
-    elif _coord_sys == 1:
-        geometry_dim = 'rz'
-    else:
-        raise Exception('Undefined coordinate system %d'%_coord_sys)
-    del _prob_lo, _coord_sys
-
 if platform.system() == 'Windows':
     path_libc = _find_library('msvcrt')
 else:
     path_libc = _find_library('c')
 _libc = ctypes.CDLL(path_libc)
+_LP_c_int = ctypes.POINTER(ctypes.c_int)
+_LP_c_char = ctypes.POINTER(ctypes.c_char)
 
-# this is a plain C/C++ shared library, not a Python module
-if os.name == 'nt':
-    mod_ext = "dll"
-else:
-    mod_ext = "so"
-libname = "libwarpx.{0}.{1}".format(geometry_dim, mod_ext)
-
-try:
-    libwarpx = ctypes.CDLL(os.path.join(_get_package_root(), libname))
-except OSError as e:
-    value = e.args[0]
-    if f'{libname}: cannot open shared object file: No such file or directory' in value:
-        raise Exception(f'"{libname}" was not installed. Installation instructions can be found here https://warpx.readthedocs.io/en/latest/install/users.html') from e
-    else:
-        print("Failed to load the libwarpx shared object library")
-        raise
+class LibWarpX():
+    def __init__(self):
+        self.__dict__['library_loaded'] = False
```

Comment:
Out of curiosity, why are these attributes set using `self.__dict__["attrib"]` rather than `self.attrib`.

Reply:
Fixed